lib/rules/block-opening-brace-space-before/__tests__/index.js

```diff
@@ -80,89 +80,103 @@ testRule(rule, {
 	],
 });
 
-testRule(rule, {
-	ruleName,
-	config: ['always', { ignoreAtRules: ['for'] }],
-	fix: true,
+for (const selector of ['for', '/fo/', /fo/]) {
+	testRule(rule, {
+		ruleName,
+		config: ['always', { ignoreAtRules: [selector] }],
+		fix: true,
 
-	accept: [
-		{
-			code: 'a { color: pink; }',
-		},
-		{
-			code: '@for ...\n{ color: pink; }',
-		},
-		{
-			code: '@for ...\r\n{ color: pink; }',
-		},
-	],
+		accept: [
+			{
+				code: 'a { color: pink; }',
+			},
+			{
+				code: '@for ...\n{ color: pink; }',
+			},
+			{
+				code: '@for ...\r\n{ color: pink; }',
+			},
+		],
 
-	reject: [
-		{
-			code: 'a{ color: pink; }',
-			fixed: 'a { color: pink; }',
-			message: messages.expectedBefore(),
-			line: 1,
-			column: 1,
-		},
-	],
-});
+		reject: [
+			{
+				code: 'a{ color: pink; }',
+				fixed: 'a { color: pink; }',
+				message: messages.expectedBefore(),
+				line: 1,
+				column: 1,
+			},
+		],
+	});
+}
 
-testRule(rule, {
-	ruleName,
-	config: ['always', { ignoreAtRules: '/fo/' }],
-	fix: true,
+for (const selector of ['a', '/a/', /a/]) {
+	testRule(rule, {
+		ruleName,
+		config: ['always', { ignoreSelectors: [selector] }],
+		fix: true,
 
-	accept: [
-		{
-			code: 'a { color: pink; }',
-		},
-		{
-			code: '@for ...\n{ color: pink; }',
-		},
-		{
-			code: '@for ...\r\n{ color: pink; }',
-		},
-	],
+		accept: [
+			{
+				code: 'a { color: pink; }',
+			},
+			{
+				code: 'a{ color: pink; }',
+			},
+			{
+				code: 'a\n{ color: pink; }',
+			},
+			{
+				code: 'a\r\n{ color: pink; }',
+			},
+		],
 
-	reject: [
-		{
-			code: 'a{ color: pink; }',
-			fixed: 'a { color: pink; }',
-			message: messages.expectedBefore(),
-			line: 1,
-			column: 1,
-		},
-	],
-});
-
-testRule(rule, {
-	ruleName,
-	config: ['always', { ignoreAtRules: /fo/ }],
-	fix: true,
-
-	accept: [
-		{
-			code: 'a { color: pink; }',
-		},
-		{
-			code: '@for ...\n{ color: pink; }',
-		},
-		{
-			code: '@for ...\r\n{ color: pink; }',
-		},
-	],
-
-	reject: [
-		{
-			code: 'a{ color: pink; }',
-			fixed: 'a { color: pink; }',
-			message: messages.expectedBefore(),
-			line: 1,
-			column: 1,
-		},
-	],
-});
+		reject: [
+			{
```

Comment:
I think a single reject test will suffice to show that the option is working as the other parts, e.g. the `optionsMatches` function, are tested elsewhere. We can use the first test, using `b{`.